persistent/Database/Persist/RDBMS/Utils.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | This module contains utility function which works on the RDBMS backends of persistent. i.e. MySQL, Postgres and Sqlite.
+--
+-- * Reference Schema & Dataset
+--
+-- All the combinators present here will be explained based on this schema:
+--
+-- > share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+-- > User
+-- >     name String
+-- >     age Int
+-- >     deriving Show
+-- > |]
+--
+-- and this dataset. The examples below will refer to this as dataset-1.
+--
+-- #dataset#
+--
+-- > +-----+-----+-----+
+-- > |id   |name |age  |
+-- > +-----+-----+-----+
+-- > |1    |SPJ  |40   |
+-- > +-----+-----+-----+
+-- > |2    |Simon|41   |
+-- > +-----+-----+-----+
+--
+module Database.Persist.RDBMS.Utils
+  ( deleteTable
+  ) where
+
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Trans.Reader
+import Database.Persist
+
+-- | Removes all the rows in the existing table. Issues \"DELETE FROM table_name\" command to the database.
+--
+-- @
+-- deleteUserRows :: MonadIO m => ReaderT SqlBackend m ()
+-- deleteUserRows = deleteTable (undefined :: User)
+-- @
+deleteTable
+  :: (PersistEntityBackend record ~ BaseBackend backend
+     ,PersistEntity record
+     ,PersistQueryWrite backend
+     ,MonadIO m)
+  => record -> ReaderT backend m ()
+deleteTable entity = deleteWhere filts
```

Comment:
This can be `deleteWhere ([] :: [Filter record])` if you enable `ScopedTypeVariables` and do `forall record backend m.` before the constraint :smile: